packages/contract/src/test-utils/test-node-launcher.ts

```diff
@@ -0,0 +1,131 @@
+import type { JsonAbi } from '@fuel-ts/abi-coder';
+import { FuelError } from '@fuel-ts/errors';
+import type { Contract } from '@fuel-ts/program';
+import type { Provider } from '@fuel-ts/providers';
+import { getForcProject } from '@fuel-ts/utils/test-utils';
+import type { WalletUnlocked } from '@fuel-ts/wallet';
+import type { LaunchCustomProviderAndGetWalletsOptions } from '@fuel-ts/wallet/test-utils';
+import { launchCustomProviderAndGetWallets } from '@fuel-ts/wallet/test-utils';
+
+import type { DeployContractOptions } from '../contract-factory';
+import ContractFactory from '../contract-factory';
+
+interface DeployContractConfig {
+  /**
+   * Directory of contract to be deployed. The sway program **must** be built beforehand.
+   */
+  contractDir: string;
+  /**
+   * Index of wallet to be used for deployment. Defaults to `0` (first wallet).
+   */
+  walletIndex?: number;
+  /**
+   * Options for contract deployment taken from `ContractFactory`.
+   */
+  options?: DeployContractOptions;
+}
+
+interface TestNodeLauncherOptions extends LaunchCustomProviderAndGetWalletsOptions {
+  deployContracts: DeployContractConfig[];
+}
+
+interface TestNodeLauncherReturn<TContracts> {
+  wallets: WalletUnlocked[];
+  provider: Provider;
+  contracts: TContracts;
+}
+
+export class TestNodeLauncher {
+  static async launch<
+    TContracts extends Contract[] = Contract[],
+    Dispose extends boolean = true,
+    ReturnType = TestNodeLauncherReturn<TContracts> &
+      (Dispose extends true ? AsyncDisposable : { cleanup: () => Promise<void> }),
+  >(
+    {
+      providerOptions = {},
+      walletConfig,
+      nodeOptions = {},
+      deployContracts = [],
+    }: Partial<TestNodeLauncherOptions> = {},
+    dispose?: Dispose
+  ): Promise<ReturnType> {
+    const { provider, wallets, cleanup } = await launchCustomProviderAndGetWallets(
+      {
+        walletConfig,
+        providerOptions,
+        nodeOptions,
+      },
+      false
+    );
+
+    try {
+      const contracts = await TestNodeLauncher.deployContracts(deployContracts, wallets);
+
+      return (
+        dispose ?? true
```

Comment:
The default is to return an object which can be disposed via `await using`. If the user passes in `dispose = false`, they won't be able to use `await using` and will have access to the `cleanup` function.